control-plane/p4RuntimeSerializer.cpp

```diff
@@ -639,6 +639,44 @@ getMatchType(cstring matchTypeName) {
     }
 }
 
+/*
+ * This function supports a deeply nested mix of Type_Newtype and typedef.
+ *The function also supported Type_Bits.
+*/
+static uint32_t
+getTypeWidth(const IR::Type* type, TypeMap* typeMap, ReferenceMap* refMap) {
```

Comment:
sorry but I'm still very dubious that this function has to be so complicated
I'm going to give this code a try and see if there is any alternative

Reply:
ok. Please use `testdata/p4_16_samples/p416-type-use3.p4` to test.

Reply:
There is subsequent work to deal with a few more issues.  In that work, I can look into how to integrate with new sizeInBits().

Reply:
I checked in code today that adds copious LOG3 message to getTypeWidth function.  The messages are displayed when this test is run.  Please analyze log and see for yourself why the code is written in this manner.  

```
./p4test --std p4-16 -T p4RuntimeSerializer.cpp:3 ../testdata/p4_16_samples/p416-type-use3.p4 --p4runtime-files tmp.txt
```

You can download the code from this PR's repo:

https://github.com/familysingh/p4c